### ISSUE TITLE
whitelist urls

### ISSUE THREAD

**Issue by jaikishantulswani** [xnl-h4ck3r/XnlReveal#1]
@xnl-h4ck3r It would be more useful if it would have a feature to add urls into whitelist category so they can be skipped.

**Comment by xnl-h4ck3r** [comment]
Hey @jaikishantulswani. Thanks for taking he time to raise an issuse, and use the extension!
I assumje you are talking about this?
```
// Define a global constant for ignored src strings
const IGNORED_STRINGS = "googletagmanager|doubleclick|google-analytics";
```
If so, you're absolutely right. It's on the TODO list, but I justed wanted to get it out there first before I get chance to do that 👍
If you're referring to something else, let me know

**Comment by jaikishantulswani** [comment]
Hi @xnl-h4ck3r my concern is what if user want to avoid testing on some urls like google.com, gmail.com or any other akamai waf enable resources. If it would have a feature where user can add their choice of urls to avoid testing on them.

**Comment by xnl-h4ck3r** [comment]
Hey. The "googletagmanager|doubleclick|google-analytics"; list is just to not try to enable or show elements where the src contains those domains. It isn't actively testing on thise domains or anything.
Also, with the call to Wayback archive, the API is only called for web pages visited... it doesn't call for other reasources on the page.
So there will be nothing "active" happening against urls like google.com, gmail.com unless you are actually on googlre.com or gmail.com itself.
Does that address your concerns?

**Comment by jaikishantulswani** [comment]
@xnl-h4ck3r I would like to make active this extension continue sly and along with this I want xnlreveal to avoid testing whenever I am on google.com,gmail.com or any other my choice of url (to avoid testing If I am on that resource)

**Comment by xnl-h4ck3r** [comment]
Oh I see, I understand now! Yes, that would make sense to have that. I guess the best thing to do would to have a list of domains that you WANT reveal enabled for, rather than supplying all you don't. So have a whitelist of domains on the options page. Does that sound good?

**Comment by jaikishantulswani** [comment]
Yes @xnl-h4ck3r this is what I mean but add a feature to blacklist some of domains which I don't want to scan instead of whitelist.

**Comment by xnl-h4ck3r** [comment]
> Yes @xnl-h4ck3r this is what I mean but add a feature to blacklist some of domains which I don't want to scan instead of whitelist.

Ok, I'll add it to the TODO lists. Thanks for the suggestion 🤘

**Comment by xnl-h4ck3r** [comment]
Hi @jaikishantulswani , this is included in v2.0 now if you want to try it out! Thanks
